File: app/src/main/java/com/mnj/java_solutions/gfg/NumberOfOneBits.java
package com.mnj.java_solutions.gfg;

public class NumberOfOneBits {

    public void setBits() {

        int n = 10;
        int count =0;

        while(n !=0) {

            int r = n % 2;

            if(r ==1)  {
                count++;
            }
            n = n/2;
        }

        System.out.println("==>> Number of 1 bits:"+count);
    }
}

File: app/src/main/java/com/mnj/java_solutions/gfg/NumberOfOneBits.kt
package com.mnj.java_solutions.gfg

class NumberOfOneBits {
    fun setBits() {
        var n = 10
        var count = 0
        while (n != 0) {
            val r = n % 2
            if (r == 1) {
                count++
            }
            n /= 2
        }
        println("==>> Number of 1 bits:$count")
    }
}
